eclair-core/src/main/scala/fr/acinq/eclair/blockchain/watchdogs/BlockchainWatchdog.scala

```diff
@@ -66,10 +67,11 @@ object BlockchainWatchdog {
             timers.startSingleTimer(CheckLatestHeaders(blockCount), delay)
             Behaviors.same
           case CheckLatestHeaders(blockCount) =>
-            context.spawn(HeadersOverDns(chainHash, blockCount), HeadersOverDns.Source) ! HeadersOverDns.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.BlockstreamExplorer()), "blockstream") ! ExplorerApi.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.BlockcypherExplorer()), "blockcypher") ! ExplorerApi.CheckLatestHeaders(context.self)
-            context.spawn(ExplorerApi(chainHash, blockCount, ExplorerApi.MempoolSpaceExplorer()), "mempool.space") ! ExplorerApi.CheckLatestHeaders(context.self)
+            val id = UUID.randomUUID()
```

Comment:
A thread-safe counter (`AtomicInteger`) could also have been used.

Reply:
True, but it's stateful and needs locks :)